tests/arch/riscv/pmp/null-pointer/src/main.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2025 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_internal.h>
+#include <zephyr/tc_util.h>
+#include <zephyr/ztest.h>
+#include <zephyr/arch/riscv/pmp.h>
+
+ZTEST_BMEM volatile bool valid_fault;
```

Comment:
No user thread use `valid_fault`, the declaration could be simplify to `static volatile bool valid_fault;`.